File: examples/governance_static_go/identity_static_go/domain/recorded_entry.go
package domain

import "time"

type RecordedEntry struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	EntityType string `json:"entity_type"`
	EntityId string `json:"entity_id"`
	Action string `json:"action"`
	ActorId string `json:"actor_id"`
	Details string `json:"details"`
	Timestamp time.Time `json:"timestamp"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e RecordedEntry) EventName() string { return "RecordedEntry" }

func (e RecordedEntry) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/identity_domain/lib/identity_domain/audit_log/events/recorded_entry.rb
module IdentityDomain
  class AuditLog
    module Events
      class RecordedEntry
        attr_reader :aggregate_id, :entity_type, :entity_id, :action, :actor_id, :details, :timestamp, :occurred_at

        def initialize(aggregate_id: nil, entity_type: nil, entity_id: nil, action: nil, actor_id: nil, details: nil, timestamp: nil)
          @aggregate_id = aggregate_id
          @entity_type = entity_type
          @entity_id = entity_id
          @action = action
          @actor_id = actor_id
          @details = details
          @timestamp = timestamp
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
